chainer/utils/__init__.py

```diff
@@ -34,3 +38,13 @@ def force_type(dtype, value):
         return value.astype(dtype, copy=False)
     else:
         return value
+
+
+@contextlib.contextmanager
+def tempdir(**kwargs):
+    # A context manager that defines a lifetime of a temporary directory.
+    temp_dir = tempfile.mkdtemp(**kwargs)
+    try:
+        yield temp_dir
+    finally:
+        shutil.rmtree(temp_dir, ignore_errors=True)
```

Comment:
Is it safe to ignore errors? (c.f. `tempfile.TemporaryDirectory` raise error when failure of cleanup: https://github.com/python/cpython/blob/3.6/Lib/tempfile.py#L777-L811)

Reply:
I agree.
I made it optional and `False` by default.